keepsorted/options.go

```diff
@@ -205,6 +240,30 @@ func parseValueWithDefault(f reflect.StructField, key, val string, defaultFn fun
 	panic(fmt.Errorf("unsupported blockOptions type: %v", f.Type))
 }
 
+func splitStringVal(val string) []string {
+	runes := []rune(val)
+	var vals []string
+	var cur strings.Builder
+	for i := 0; i < len(runes); i++ {
+		r := runes[i]
+		switch r {
+		case ',':
+			vals = append(vals, cur.String())
+			cur.Reset()
+		case '\\':
+			if i+1 < len(runes) && runes[i+1] == ',' {
```

Comment:
Optional: Rather than escaping the commas (confusing, since no other language does that), why don't we flip things slightly. Rather than first parsing all options as strings, we decide the type of the key and parse the option according to the key type. We actually don't have pure string options. They're all either `map[string]bool` or `[]string`. Rather than the syntax being like `ignore_prefixes="a,b c,d\,e"`, it perhaps is more idiomatic to have it be `ignore_prefixes=a,"b c","d,e"`. That's very close to what most languages would do, except we're missing the typical square brackets that would allow spaces between commas (e.g., `ignore_prefixes=[a, "b c", "d,e"]` -- valid YAML!). We _could_ optionally support such square braces? I mean, it's pretty comparable in how much of a breaking change it is to how the PR currently is.

I recognize this drastically changes your PR, though. I'm not inclined to be a stickler for it, but do think this is a more intuitive syntax that gives us more future flexibility, too. The annoying thing is that for backwards compatibility, we still have to support the no square brace format, but we could treat that as a super basic "no whitespace and no escaping" case for ease of handling (i.e., as it currently is).